File: 2/2.scala

def isSorted[T](array: List[T], isSorted: (T, T) => Boolean) : Boolean =
    (array, array.tail).zipped map isSorted forall identity

def intSorted(x: Int, y : Int) : Boolean = x <= y

def intSorted2(x: Int, y : Int) : Boolean = x >= y

val x : Any

x match {
  case y:String => println(y)
  case y:Int => println(y)
}

File: 2/2.hs

isSorted :: [a] -> (a -> a -> Bool) -> Bool
isSorted l cmp = and $ zipWith cmp l (tail l)

main :: IO ()
main = do print $ isSorted [1..100] (<=)
          print $ isSorted [7,38,2,1,0,99] (<=)
          print $ isSorted [7,38,2,1,0,99] (const $ const True)
          print $ isSorted [100,99..1] (<=)
          print $ isSorted [100,99..1] (>=)
          print $ isSorted ['a'..'z'] (<=)
